v2/spiffeid/id.go

```diff
@@ -1,78 +1,163 @@
 package spiffeid
 
 import (
+	"errors"
+	"fmt"
 	"net/url"
+	"strings"
 )
 
 // ID is a SPIFFE ID
 type ID struct {
+	uri *url.URL
```

Comment:
While reviewing the bundle set, I realized that implementing this using an embedded `*url.URL` is going to do the wrong thing when using the `ID` type as a key in a map. I think this should be implemented as follows:
```
type ID struct {
    td TrustDomain
    path string
}
```

We can still use url.Parse to facilitate parsing and use `url.URL` when rendering `String()`.